### PR TITLE
[WIP]: make browser tab should open via Node API

### PR DESCRIPTION
**Pull request by hiroppy**
closes: #1510

- [x] This is a **bugfix**
- [ ] This is a **feature**
- [ ] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **docs update**
- [ ] This is a **metadata update**

### Motivation / Use-Case

Migrate migrate `webpack-dev-server.js`'s listen function to `server.js`.

### Breaking Changes

no

### Additional Info

Wait for #1658

### CODE REVIEW

File: lib/Server.js [hunk 1]
```diff
@@ -81,861 +83,928 @@ const STATS = {
...
+    this.contentBaseWatchers.push(watcher);
   }
-};
+
+  invalidate() {
+    if (this.middleware) {
+      this.middleware.invalidate();
+    }
+  }
+}
```
Comment (evilebottnawi) [inline lib/Server.js]: Let's merge https://github.com/webpack/webpack-dev-server/pull/1658 first and when rebase :+1:

**Comment by hiroppy** [comment]
I am busy this week so I will fix this issue next week:)

**Pull request closed without merging**